FileEmulationFramework/FileAccessServer.cs

```diff
@@ -184,18 +184,24 @@ private static int NtReadFileImpl(IntPtr handle, IntPtr hEvent, IntPtr* apcRouti
             requestedOffset += numReadBytes;
             SetInformationFileImpl(handle, ioStatus, (byte*)&requestedOffset, sizeof(long), FileInformationClass.FilePositionInformation);
 
-            // Set number of read bytes.
-            ioStatus->Status = 0;
+            // Set status
+            ioStatus->Status = NT_STATUS.STATUS_SUCCESS;
             ioStatus->Information = new(numReadBytes);
-            return 0;
+            return NT_STATUS.STATUS_SUCCESS;
+        }
+        else
+        {
+            _logger.Debug("[FileAccessServer] Read Fail, Length: {0}, Offset: {1}", numReadBytes, requestedOffset);
```

Comment:
I'm fine with the change; this should say `likely EOF` though in brackets though.
The documentation for `ReadData` in the interface also needs updating.

Since this is (technically) a breaking change, we'd need to check with @LTSophia and @morgana-x before merging. I believe @LTSophia implemented support for some of the less commonly used games (e.g. JSR), and @morgana-x did danganrompa

Reply:
For the `ReadData` documentation does this sound fine as the return value?
> True if at least one byte was read, false otherwise (this indicates the EOF has been reached).

Reply:
This sounds good to me.

In the Rust version, we gotta make sure that EOF is more explicitly defined.

If there's a real error (read fail), we would panic there in that case.